src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1584,7 +1589,10 @@ internal void SetRequestContent(HttpRequestMessage request, IDictionary content)
 
         internal void ParseLinkHeader(HttpResponseMessage response)
         {
-            Uri requestUri = response.RequestMessage.RequestUri;
+            Uri? requestUri = response.RequestMessage?.RequestUri;
+
+            ArgumentNullException.ThrowIfNull(requestUri);
```

Comment:
What is the scenario? Why should we throw? What is an expected behavior?

Reply:
Looking how the method works I guess it is not correct. If we cannot parse link header I'd expect we just return.